TrackerMC/inc/StrawDigiBundle.hh

```diff
@@ -0,0 +1,35 @@
+// Bundle together references to various StrawDigi products, to circumvent
+// triplicated-loops which pop up when iterating over digis
+//
+// Ed Callaghan, 2023
+
+#ifndef TrackerMC_StrawDigiBundle_hh
+#define TrackerMC_StrawDigiBundle_hh
+
+// mu2e
+#include <Offline/RecoDataProducts/inc/StrawDigi.hh>
+#include <Offline/MCDataProducts/inc/StrawDigiMC.hh>
+
+namespace mu2e{
+  class StrawDigiBundle{
+    public:
+      StrawDigiBundle(const StrawDigi digi, const StrawDigiADCWaveform adcs, const StrawDigiMC mc);
+      StrawDigiBundle(const StrawDigi digi, const StrawDigiADCWaveform adcs);
+      StrawDigiBundle(const StrawDigiBundle& bundle);
+
+      const StrawDigi GetStrawDigi() const;
+      const StrawDigiADCWaveform GetStrawDigiADCWaveform() const;
+      const StrawDigiMC GetStrawDigiMC() const;
+
+      // interface for sorting into buckets of overlapping digitization windows
+      const double time() const;
+    protected:
+      const StrawDigi digi;
```

Comment:
There are two reasons that the members are values instead of references. The first is that, to use references, the underlying types (specifically the waveforms, IIRC) would need non-trivial copy constructors implemented and some of the loops need to be restructured to deal with the limitations of constructing objects with reference members. Those are hurdles which are both completely surmountable and, IMO, annoying.

The second reason is that, while right now all of the digis do come "from an event," this will not be true in the future: when we implement "real" collision resolution, e.g. by redigitizing a reconstructed analog sum, then this collection will be responsible for transferring entirely new data products into the scope of the simulation module, and hence should be deep-copied. But as @rlcee pointed out above, the getters can at least return references.

Reply:
Looking ahead, will some of these data members be purely input and others output?  If so, I suggest holding inputs by const& and then outputs must be by value.

Reply:
They will all be used bidirectionally. That could change if, in the future, the collision resolution is factored out of `StrawDigisFromStrawGasSteps` and into some standalone module.